### DISCUSSION TITLE
Табличное отображение в панели

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#67] [Как это сделать?] [answered]
Здравствуйте.
Часто нужно отобразить какие то табличные данные в панели ячейки.
Например, список товаров в заказе, не заходя в сам цикл. 
Как это можно сделать в таком формате, как на картинке?
![image](https://user-images.githubusercontent.com/19141501/151864470-930b2755-0a43-4d3b-9daa-22f5bb31a1c6.png)

Нашел на эту тему пример в типе данные https://docs.totum.online/fields#type-listRow, но честно говоря ничего не понял)

**Comment by erpic-ru** [comment, marked as answer]
В нужное поле в таблице циклов добавить код форматирования: 
`p1=: panelHtml(html: $list)`

Можно взять данные из цикла и вперед:
`data: selectlist(table: '...'; cycle: #id; field: '.....'; order: 'id' asc)`

Каждая запись с новой строки:
`list: listJoin(list: $data; str: $#nl)`

Можно добавить html-верстку и получить то, что у вас на картинке

[Документация](https://docs.totum.online/functions#fn-panelHtml)

**Reply by ttm-alexeypolunin** [reply]
То, что показано на картинке это специальный функционал поля тип Данные. Если определенным образом оформить в нем rowList, то он будет отображаться в контекстной панели таким образом. 

Каким образом выглядит шаблон:

<img width="452" alt="Снимок экрана 2022-02-04 в 14 53 51" src="https://user-images.githubusercontent.com/55755565/152524926-193e0de1-fbbf-4826-ab9f-2e4eaa1aaf6c.png">


```json
{
  "data": [
    {
      "name_1": "content_here_1_1",
      "name_2": "content_here_1_2",
      "name_3": true
    },
    {
      "name_1": "content_here_2_1",
      "name_2": "content_here_2_2",
      "name_3": false
    }
  ],
  "settings": {
    "columns": [
      "name_1",
      "name_2",
      "name_3"
    ],
    "headRow": {
      "name_1": "Title for name_1",
      "name_2": "Title for name_2",
      "name_3": "Title for name_3"
    },
    "headColumn": true
  }
}
```

В `data` содержаться данные для показа — rowList. Строки это строки, колонки это колонки. Там может быть больше колонок чем будет показано. Показываться будут те, которые указаны в секции `settings` — `columns`.

В `settings` оформляются настройки отображения:

+ `columns` — `name` колонок из `data`, которые надо показать.

+ `headRow` — строка заголовков в которой мы делаем `row` из `name` колонки и соответствующего ей `title`.

+ `headColumn` — `true` | `false`, нужно ли выделять жирным значения в первой колонке.

Что бы в поле значение выглядело прилично делается `setFormat(text: "...")`.

Шаблон можно собрать прям в поле кодом, либо хранить где-то и подтягивать селектом.

Я обычно имею в таблице два поля — техническое с данными `data_from_teh_field` и показываемое пользователю:

```
=: rowCreate(field: "data" = #data_from_teh_field; field: "settings" = $settings)
    settings: rowCreate(field: "columns" = $columns; field: "headRow" = $headRow; field: "headColumn" = true)
        columns: json`["name_1","tip_licenzii","name_2","name_3"]`
    headRow: rowCreate(field: "name_1" = "Title 1"; field: "name_2" = "Title 2"; field: "name_3" = "Title 3")

```

Еще можно сделать этому полю контекстную панель пошире — [https://docs.totum.online/fields#fields-settings-contextPanelWidth](https://docs.totum.online/fields#fields-settings-contextPanelWidth)
